### ISSUE TITLE
Lint/SafeNavigationChain shouldn't complain about chaining `to_d`

### ISSUE THREAD

**Issue by easybills-admin** [rubocop-hq/rubocop#6633]
## Expected behavior

`nil.to_d` returns `0.0`, so Rubocop should **not** complain about this case in particular.

## Actual behavior

It complains:
```
# Lint/SafeNavigationChain: Do not chain ordinary method call after safe navigation operator.
```

**Note:** Rubocop doesn't complain about the  same code with `to_f`.

## Steps to reproduce the problem

Just write the following code and run Rubocop:

```ruby
amount&.slice.to_d
```

## RuboCop version

```
0.62.0 (using Parser 2.5.3.0, running on ruby 2.6.0 x86_64-darwin18)
```
